services/av_worker.js

```diff
@@ -9,22 +9,31 @@ const https = require("https");
 const {json, send} = micro;
 const config = require("./av.config");
 
-const server = micro(async req => {
-    const data = json(req);
+const server = micro(async (req, res) => {
+    let uwu = json(req);
+    try {
+        let vtRes = await request("POST", `https://www.virustotal.com/vtapi/v2/url/report`, {headers: {"Content-Type": "appplication/json", "Accept-Encoding": "gzip, deflate"}}, JSON.stringify({resource: uwu.url, apikey: config.apiKey, scan: "1"}));
 
-    request("POST", `https://www.virustotal.com/vtapi/v2/url/scan?apikey=${config.apiKey}`, {}, {data}).then(res => {
+        if (vtRes !== "204") send(res, 429, JSON.stringify({code: "429", message: "Exceeded Virustotal ratelimit."}));
+        else {
 
-        if (res.statusCode === "204") send(res, 204, JSON.stringify({code: "204", reason: "Exceeded VirusTotal Ratelimit."}));
-        // TODO: Check to VT if its a safe file.
-        // because this one returns only a permalink.
-        else send(res, 200, JSON.stringify(res));
-    });
+            // Scans the URL. We inform the front-facing REST if its safe or not.
+
+            let data = JSON.parse(vtRes).body;
+
+            if (data.scans.MalwarePatrol.detected !== "false") send(res, 200, JSON.stringify({code: "200", is_bad: true, message: "Reported URL Malware. Delete immediately."}));
+
+            else send(res, 200, JSON.stringify({code: "200", is_bad: false, message: "URL Reported OK. No Malware found."}));
+        }
+    } catch(err) {
+        send(res, 500, JSON.stringify({message: `Could not fulfill request. ${err}`}));
```

Comment:
Maybe just pass the original error along, instead of putting it in a string with some other stuff?